lib/src/test/kotlin/starknet/provider/ProviderTest.kt

```diff
@@ -327,29 +320,110 @@ class ProviderTest {
+                                "output_builtin": 0,
+                                "pedersen_builtin": 0,
+                                "range_check_builtin": 0
+                            },
+                            "n_memory_holes": 0,
+                            "n_steps": 41
+                        },
+                        "l2_to_l1_messages": [],
+                        "status": "ACCEPTED_ON_L1",
+                        "transaction_hash": "0x1a9d9e311ff31e27b20a7919bec6861dd6b603d72b7e8df9900cd7603200d0b",
+                        "transaction_index": 8
+                    }
+                """.trimIndent(),
+            )
+        }
+        val provider = GatewayProvider("", "", StarknetChainId.TESTNET, httpService)
+
+        val request = provider.getTransactionReceipt(Felt.ZERO)
         val response = request.send()
 
-        assertNotNull(response)
         assertTrue(response is GatewayTransactionReceipt)
     }
 
     @Test
-    fun `get declare transaction receipt gateway`() {
-        val request = gatewayProvider().getTransactionReceipt(declareTransactionHash)
+    fun `get deploy transaction receipt rpc`() {
+        val httpService = mock<HttpService> {
+            on { send(any()) } doReturn HttpResponse(
+                true,
+                200,
+                """
+                    {
+                        "id": 0,
+                        "jsonrpc": "2.0",
+                        "result":
+                        {
+                            "actual_fee": "0x0",
+                            "block_hash": "0x4e782152c52c8637e03df60048deb4f6adf122ef37cf53eeb72322a4b9c9c52",
+                            "contract_address": "0x20f8c63faff27a0c5fe8a25dc1635c40c971bf67b8c35c6089a998649dfdfcb",
+                            "transaction_hash": "0x1a9d9e311ff31e27b20a7919bec6861dd6b603d72b7e8df9900cd7603200d0b",
+                            "status": "ACCEPTED_ON_L1",
+                            "block_number": 264715,
+                            "type": "DEPLOY",
+                            "events":
+                            [],
+                            "messages_sent":
+                            []
+                        }
+                    }
+                """.trimIndent(),
+            )
+        }
+        val provider = JsonRpcProvider(devnetClient.rpcUrl, StarknetChainId.TESTNET, httpService)
+
+        val request = provider.getTransactionReceipt(Felt.ZERO)
+        val response = request.send()
+
+        assertTrue(response is DeployRpcTransactionReceipt)
+    }
+
+    @ParameterizedTest
+    @MethodSource("getProviders")
+    fun `get declare transaction receipt`(provider: Provider) {
+        val request = provider.getTransactionReceipt(declareTransactionHash)
         val response = request.send()
 
         assertNotNull(response)
-        assertTrue(response is GatewayTransactionReceipt)
+
+        if (provider is GatewayProvider) {
```

Comment:
Is there not some base class for all receipts that we could use instead? Or did you explicitly want to check for GatewayTransactionReceipt and RpcTransactionReceipt?

Reply:
There is, I wanted to make it explicit. There is a slight possibility that wrong receipt type would be returned (pending/deploy/pendingDeploy  - and if that happens I think the test should fail because something is wrong)